apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/sidebar-modal-opener.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { Button, Modal } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { PageTemplatesPlugin } from '../index';
+import TemplateSelectorItem from './template-selector-item';
+import replacePlaceholders from '../utils/replace-placeholders';
+/* eslint-enable import/no-extraneous-dependencies */
+
+class SidebarModalOpener extends Component {
+	state = {
+		isTemplateModalOpen: false,
+		isWarningOpen: false,
+	};
+
+	toggleTemplateModal = () => {
+		this.setState( {
+			isTemplateModalOpen: ! this.state.isTemplateModalOpen,
+			isWarningOpen: false,
+		} );
+	};
+
+	toggleWarningModal = () => {
+		this.setState( { isWarningOpen: ! this.state.isWarningOpen } );
+	};
+
+	getLastTemplateUsed = () => {
+		if ( ! this.props.templateUsedSlug || this.props.templateUsedSlug === 'blank' ) {
+			return this.props.templates[ 1 ];
+		}
+		return this.props.templates.filter( temp => temp.slug === this.props.templateUsedSlug )[ 0 ];
+	};
+
+	// The TemplateSelectorItem wants a function to run when clicked, we want it to do nothing from here.
+	onSelectOverride = () => {};
+
+	render() {
+		const { slug, title, preview, previewAlt } = this.getLastTemplateUsed();
+		const { templates, vertical, segment, siteInformation } = this.props;
+
+		return (
+			<div className="sidebar-modal-opener">
+				<TemplateSelectorItem
+					id="sidebar-modal-opener__last-template-used-preview"
+					value={ slug }
+					label={ replacePlaceholders( title, siteInformation ) }
+					staticPreviewImg={ preview }
+					staticPreviewImgAlt={ previewAlt }
+					onSelect={ this.onSelectOverride }
```

Comment:
This leads me to think that maybe we should use a different component for this?

For example, it feels like this should be clickable, but it isn't.

![2019-11-12 12 24 32](https://user-images.githubusercontent.com/6265975/68708083-d609ba00-0547-11ea-85b6-ec3b9b0038c8.gif)

Though, I wonder if maybe `onSelect` could also open the modal :)

Reply:
I also feel like the styles seem a bit off compared to the selector. I wonder if we should try to match those

<img width="170" alt="Screen Shot 2019-11-12 at 12 41 22 PM" src="https://user-images.githubusercontent.com/6265975/68709064-c7240700-0549-11ea-9c76-eb263f34845e.png">

Reply:
> it feels like this should be clickable, but it isn't.

Whoah, I do not notice that behavior in my browser.  But I agree that having that click open the warning modal is a great idea.  Updated!

> I also feel like the styles seem a bit off compared to the selector. I wonder if we should try to match those

Perhaps we should?  At least I agree the label at the bottom should be a bit better.

Currently the style in the sidebar renders the full length preview of the preview image, which is fully representative of the template.  The fullscreen selector does limit all of their height so they list well in a group.  Im not exactly sure which way we should go with that height property, but it is worth discussing further!

Reply:
cc @shaunandrews / @olaolusoga for thoughts on styling :)